### PR TITLE
Issue: #4091: Changing info logs to warn/error for download failure

### PR DESCRIPTION
**Pull request by rutujasurve94**
Changing info logs to warn/error on download failure for sample data.

### CODE REVIEW

File: changelog/4149.trivialfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed info logs for download sample failure.
```
Comment (nabobalis) [inline changelog/4149.trivialfix.rst:1]: Can you remove this changelog please?

Comment (rutujasurve94) [reply changelog/4149.trivialfix.rst]: done.

File: sunpy/data/_sample.py [hunk 2]
```diff
@@ -104,7 +104,7 @@ def download_sample_data(overwrite=False):
     if not results.errors:
         return results
     else:
-        log.info('Failed to download sample data, retrying with a mirror.')
+        warnings.warn('Failed to download sample data, retrying with a mirror.')
```
Comment (Cadair) [inline sunpy/data/_sample.py:107]: My instinct here is `info` is correct. I feel like `warning` implies the user can and should do something about it, this is more likely to be a server error for us. We want it to print so we can see and fix those errors.

File: sunpy/data/_sample.py [hunk 3]
```diff
@@ -122,7 +122,7 @@ def download_sample_data(overwrite=False):
 
     for err in results.errors:
         file_name = Path(err.url).name
-        log.info('Failed to download file' + str(err.filepath_partial) +
+        log.error('Failed to download file' + str(err.filepath_partial) +
                  'from url: ' + str(err.url) + ' with error: ' + str(err.exception))
         warnings.warn(f"File {file_name} not found.", SunpyUserWarning)
```
Comment (Cadair) [inline sunpy/data/_sample.py:127]: Do we need both these things here? Spewing roughly the same intel here twice at different log levels seems flawed?

**Comment by Cadair** [comment]
Thanks for the PR @rutujasurve94 It's nice to be able to discuss this with a nice simple diff :smile:

File: sunpy/data/_sample.py (hunk 3 above)
Comment (nabobalis) [reply sunpy/data/_sample.py:127]: So combine these and keep it at info?

Comment (rutujasurve94) [reply sunpy/data/_sample.py]: sure, updated the PR.

File: sunpy/data/_sample.py (hunk 2 above)
Comment (rutujasurve94) [reply sunpy/data/_sample.py]: yes sure, updated the PR

**Comment by Cadair** [comment]
@rutujasurve94 I hope you don't mind, I pushed a commit with some better wording for the messages. I have pushed the actual content of the connection error to the "debug" level, which now prints the error for all the mirrors. Without the debug a complete faliure looks like:

```
>>> import sunpy.data.sample                                                                                                                                                                        
Files Downloaded:   0%|                                                                                                                                                                                                               | 0/26 [00:00<?, ?file/s]
INFO: Failed to download one or more sample data files, retrying with a mirror. [sunpy.data._sample]
Files Downloaded:   0%|                                                                                                                                                                                                               | 0/26 [00:00<?, ?file/s]
ERROR: Failed to download CALLISTO_SPECTRUM from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download RHESSI_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download HMI_LOS_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download NOAAPREDICT_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_193_CUTOUT02_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_171_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_1600_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_094_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_335_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download NOAAINDICES_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_211_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_193_CUTOUT03_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download GOES_XRS_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download RHESSI_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download SWAP_LEVEL1_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_193_CUTOUT01_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_171_ROLL_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download LYRA_LEVEL3_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download NORH_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_193_CUTOUT04_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download EVE_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download EIT_195_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_131_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_193_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_193_CUTOUT05_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download GBM_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
```

with debug it looks like:
```
>>> import sunpy; sunpy.log.setLevel('DEBUG'); import sunpy.data.sample                                                                                                                         
Files Downloaded:   0%|                                                                                                                                                                                                               | 0/26 [00:00<?, ?file/s]
INFO: Failed to download one or more sample data files, retrying with a mirror. [sunpy.data._sample]
DEBUG: Failed to download AIA_335_IMAGE from http://data2.sunpy.org/sunpy/v1/AIA20110607_063303_0335_lowres.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download NOAAINDICES_TIMESERIES from http://data2.sunpy.org/sunpy/v1/swpc_solar_cycle_indices.txt: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_211_IMAGE from http://data2.sunpy.org/sunpy/v1/AIA20110607_063302_0211_lowres.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_193_CUTOUT03_IMAGE from http://data2.sunpy.org/sunpy/v1/AIA20110607_064555_0193_cutout.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download RHESSI_TIMESERIES from http://data2.sunpy.org/sunpy/v1/hsi_obssumm_20110607_025.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download GBM_TIMESERIES from http://data2.sunpy.org/sunpy/v1/glg_cspec_n5_110607_v00.pha: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download EIT_195_IMAGE from http://data2.sunpy.org/sunpy/v1/eit_l1_20110607_203753.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download EVE_TIMESERIES from http://data2.sunpy.org/sunpy/v1/20110607_EVE_L0CS_DIODES_1m.txt: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_171_ROLL_IMAGE from http://data2.sunpy.org/sunpy/v1/aiacalibim5.fits.gz: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download SWAP_LEVEL1_IMAGE from http://data2.sunpy.org/sunpy/v1/swap_lv1_20110607_063329.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_193_IMAGE from http://data2.sunpy.org/sunpy/v1/AIA20110607_063307_0193_lowres.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_193_CUTOUT02_IMAGE from http://data2.sunpy.org/sunpy/v1/AIA20110607_063931_0193_cutout.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download HMI_LOS_IMAGE from http://data2.sunpy.org/sunpy/v1/HMI20110607_063211_los_lowres.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download CALLISTO_SPECTRUM from http://data2.sunpy.org/sunpy/v1/BIR_20110607_062400_10.fit: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_1600_IMAGE from http://data2.sunpy.org/sunpy/v1/AIA20110607_063305_1600_lowres.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_131_IMAGE from http://data2.sunpy.org/sunpy/v1/AIA20110607_063301_0131_lowres.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download GOES_XRS_TIMESERIES from http://data2.sunpy.org/sunpy/v1/go1520110607.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_094_IMAGE from http://data2.sunpy.org/sunpy/v1/AIA20110607_063305_0094_lowres.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download LYRA_LEVEL3_TIMESERIES from http://data2.sunpy.org/sunpy/v1/lyra_20110607-000000_lev3_std.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download NORH_TIMESERIES from http://data2.sunpy.org/sunpy/v1/tca110607.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_193_CUTOUT01_IMAGE from http://data2.sunpy.org/sunpy/v1/AIA20110607_063307_0193_cutout.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_171_IMAGE from http://data2.sunpy.org/sunpy/v1/AIA20110607_063302_0171_lowres.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download RHESSI_IMAGE from http://data2.sunpy.org/sunpy/v1/hsi_image_20110607_063300.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_193_CUTOUT05_IMAGE from http://data2.sunpy.org/sunpy/v1/AIA20110607_065843_0193_cutout.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download NOAAPREDICT_TIMESERIES from http://data2.sunpy.org/sunpy/v1/predicted-sunspot-radio-flux.txt: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_193_CUTOUT04_IMAGE from http://data2.sunpy.org/sunpy/v1/AIA20110607_065219_0193_cutout.fits: Cannot connect to host data2.sunpy.org:80 ssl:default [Name or service not known] [sunpy.data._sample]
Files Downloaded:   0%|                                                                                                                                                                                                               | 0/26 [00:00<?, ?file/s]
ERROR: Failed to download CALLISTO_SPECTRUM from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download RHESSI_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download HMI_LOS_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download NOAAPREDICT_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_193_CUTOUT02_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_171_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_1600_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_094_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_335_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download NOAAINDICES_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_211_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_193_CUTOUT03_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download GOES_XRS_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download RHESSI_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download SWAP_LEVEL1_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_193_CUTOUT01_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_171_ROLL_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download LYRA_LEVEL3_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download NORH_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_193_CUTOUT04_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download EVE_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download EIT_195_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_131_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_193_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download AIA_193_CUTOUT05_IMAGE from all mirrors, the file will not be availible. [sunpy.data._sample]
ERROR: Failed to download GBM_TIMESERIES from all mirrors, the file will not be availible. [sunpy.data._sample]
DEBUG: Failed to download CALLISTO_SPECTRUM from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/BIR_20110607_062400_10.fit: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download RHESSI_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/hsi_image_20110607_063300.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download HMI_LOS_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/HMI20110607_063211_los_lowres.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download NOAAPREDICT_TIMESERIES from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/predicted-sunspot-radio-flux.txt: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_193_CUTOUT02_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/AIA20110607_063931_0193_cutout.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_171_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/AIA20110607_063302_0171_lowres.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_1600_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/AIA20110607_063305_1600_lowres.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_094_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/AIA20110607_063305_0094_lowres.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_335_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/AIA20110607_063303_0335_lowres.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download NOAAINDICES_TIMESERIES from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/swpc_solar_cycle_indices.txt: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_211_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/AIA20110607_063302_0211_lowres.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_193_CUTOUT03_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/AIA20110607_064555_0193_cutout.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download GOES_XRS_TIMESERIES from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/go1520110607.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download RHESSI_TIMESERIES from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/hsi_obssumm_20110607_025.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download SWAP_LEVEL1_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/swap_lv1_20110607_063329.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_193_CUTOUT01_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/AIA20110607_063307_0193_cutout.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_171_ROLL_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/aiacalibim5.fits.gz: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download LYRA_LEVEL3_TIMESERIES from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/lyra_20110607-000000_lev3_std.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download NORH_TIMESERIES from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/tca110607.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_193_CUTOUT04_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/AIA20110607_065219_0193_cutout.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download EVE_TIMESERIES from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/20110607_EVE_L0CS_DIODES_1m.txt: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download EIT_195_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/eit_l1_20110607_203753.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_131_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/AIA20110607_063301_0131_lowres.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_193_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/AIA20110607_063307_0193_lowres.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download AIA_193_CUTOUT05_IMAGE from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/AIA20110607_065843_0193_cutout.fits: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
DEBUG: Failed to download GBM_TIMESERIES from https://githubw.com/sunpy/sample-data/raw/master/sunpy/v1/glg_cspec_n5_110607_v00.pha: Cannot connect to host githubw.com:443 ssl:default [Name or service not known] [sunpy.data._sample]
```

**Pull request merged**